modules/gdpr_fields/src/Entity/GdprField.php

```diff
@@ -148,4 +189,207 @@ public function rtaDescription() {
+   * @return $this
+   */
+  public function setRtf($rtf) {
+    $this->rtf = $rtf;
+    return $this;
+  }
+
+  /**
+   * Sets the anonymizer ID.
+   *
+   * @param string $anonymizer
+   *   Anonymizer ID.
+   *
+   * @return $this
+   */
+  public function setAnonymizer($anonymizer) {
+    $this->anonymizer = $anonymizer;
+    return $this;
+  }
+
+  /**
+   * Sets field notes.
+   *
+   * @param string $notes
+   *   Notes.
+   *
+   * @return $this
+   */
+  public function setNotes($notes) {
+    $this->notes = $notes;
+    return $this;
+  }
+
+  /**
+   * Sets SAR filename.
+   *
+   * @param string $filename
+   *   Filename.
+   *
+   * @return $this
+   */
+  public function setSarsFilename($filename) {
+    $this->sarsFilename = $filename;
+    return $this;
+  }
+
+  /**
+   * Sets the relationship.
+   *
+   * @param int $relationship
+   *   Relationship setting.
+   *
+   * @return $this
+   */
+  public function setRelationship($relationship) {
+    $this->relationship = $relationship;
+    return $this;
+  }
+
+  /**
+   * Get the options array for right to access field.
+   *
+   * @return array
+   *   Right to access field options array.
+   */
+  public static function rtaOptions() {
+    return [
+      'inc' => 'Included',
+      'maybe' => 'Maybe included',
+      'no' => 'Not included',
+    ];
+  }
+
+  /**
+   * Get the options array for right to be forgotten field.
+   *
+   * @return array
+   *   Right to be forgotten field options array.
+   */
+  public static function rtfOptions() {
+    return [
+      'anonymize' => 'Anonymize',
+      'remove' => 'Remove',
+      'maybe' => 'Maybe included',
+      'no' => 'Not included',
+    ];
+  }
+
+  /**
+   * Check whether a property can be removed.
+   *
+   * @param \Drupal\Core\Field\FieldDefinitionInterface $field_definition
+   *   The property info.
+   * @param string $error_message
+   *   A variable to fill with an error message.
+   *
+   * @return bool
+   *   TRUE if the property can be removed, FALSE if not.
+   */
+  public function propertyCanBeRemoved(FieldDefinitionInterface $field_definition, &$error_message = NULL) {
+    if ($field_definition->isComputed()) {
+      $error_message = new TranslatableMarkup('Unable to remove computed field %field.', ['%field' => $field_definition->getName()]);
+      return FALSE;
+    }
+
+    if ($field_definition->isRequired()) {
+      $error_message = new TranslatableMarkup('Unable to remove required field %field.', ['%field' => $field_definition->getName()]);
+      return FALSE;
+    }
+
+    if ($field_definition->isReadOnly()) {
+      $error_message = new TranslatableMarkup('Unable to remove readonly field %field.', ['%field' => $field_definition->getName()]);
+      return FALSE;
+    }
+
+    $entity_type = \Drupal::entityTypeManager()->getDefinition($this->entityTypeId);
+    // @todo Find something less generic than `getKeys()`.
+    if (in_array($field_definition->getName(), $entity_type->getKeys())) {
```

Comment:
Should we just hard-code an array of known keys for now?

Reply:
@andrewbelcher ?

Reply:
I think a hardcoded set of keys is fine. From memory, I would suggest:

- `id`
- `uuid`
- `revision`
- `bundle`
- `language`
- `status`?
- `uid`?

Thinks of any others that are mega important?